src/rnp/rnpcfg.c

```diff
@@ -497,6 +576,105 @@ rnp_cfg_get_defkey(rnp_cfg_t *cfg, rnp_params_t *params)
     } else {
         params->defkey = strdup(userid);
     }
+}
 
-    return RNP_OK;
+/**
+ * @brief Grabs date from the string in %Y-%m-%d format
+ *
+ * @param s [in] NULL-terminated string with the date
+ * @param t [out] On successfull return result will be placed here
+ * @return true on success or false otherwise
+ */
+
+static bool
+grabdate(const char *s, int64_t *t)
+{
+    static regex_t r;
+    static int     compiled;
+    regmatch_t     matches[10];
+    struct tm      tm;
+
+    if (!compiled) {
+        compiled = 1;
+        (void) regcomp(
+          &r, "([0-9][0-9][0-9][0-9])[-/]([0-9][0-9])[-/]([0-9][0-9])", REG_EXTENDED);
+    }
+    if (regexec(&r, s, 10, matches, 0) == 0) {
+        (void) memset(&tm, 0x0, sizeof(tm));
+        tm.tm_year = (int) strtol(&s[(int) matches[1].rm_so], NULL, 10);
+        tm.tm_mon = (int) strtol(&s[(int) matches[2].rm_so], NULL, 10) - 1;
+        tm.tm_mday = (int) strtol(&s[(int) matches[3].rm_so], NULL, 10);
+        *t = mktime(&tm);
+        return true;
+    }
+    return false;
+}
+
+/**
+ * @brief Get signature validity duration time from the user input
+ *
+ * Signature duration may be specified in different formats:
+ * - 10d : 10 days (you can use [h]ours, d[ays], [w]eeks, [m]onthes)
+ * - 2017-07-12 : as the exact date when signature becomes invalid
+ * - 60000 : number of seconds
+ *
+ * @param s [in] NULL-terminated string with the date
+ * @param t [out] On successfull return result will be placed here
+ * @return duration time in seconds
+ */
+
+uint64_t
+get_duration(const char *s)
```

Comment:
@flowher Maybe we ought to discuss this over in https://github.com/riboseinc/rnp/issues/192 instead?